### PR TITLE
Withdrawal epoch transition

### PR DESCRIPTION
**Pull request by potuz**
Implement the main loop `process_withdrawals_into_queue` from https://github.com/ethereum/consensus-specs/pull/3042

**Note to reviewers:** 
- Review only if  the spec PR is approved/merged
- Review only after #11570 has merged 
- There are proto changes and breaking changes to the protobuf messages but on Capella structures that have not been used.

### CODE REVIEW

**Review by terencechain — CHANGES_REQUESTED** [review]

File: proto/prysm/v1alpha1/attestation/attestation_utils.go [hunk 1]
```diff
@@ -87,19 +89,20 @@ func AttestingIndices(bf bitfield.Bitfield, committee []types.ValidatorIndex) ([
...
+//	 """
+//	 # Verify indices are sorted and unique
+//	 indices = indexed_attestation.attesting_indices
+//	 if len(indices) == 0 or not indices == sorted(set(indices)):
+//	     return False
+//	 # Verify aggregate signature
+//	 pubkeys = [state.validators[i].pubkey for i in indices]
+//	 domain = get_domain(state, DOMAIN_BEACON_ATTESTER, indexed_attestation.data.target.epoch)
+//	 signing_root = compute_signing_root(indexed_attestation.data, domain)
+//	 return bls.FastAggregateVerify(pubkeys, signing_root, indexed_attestation.signature)
```
[the remark this answers is not in the document]
Comment (potuz) [reply proto/prysm/v1alpha1/attestation/attestation_utils.go:105]: I did not touch this file

[the remark this answers is not in the document]
Comment (potuz) [reply proto/prysm/v1alpha1/attestation/attestation_utils.go:105]: Ohh it's the last commit is bogus, just a sec

File: beacon-chain/core/capella/transition.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			return nil, errors.Wrapf(err, "could not get balance of validator at index %d", index)
+		}
+		if val.IsFullyWithdrawable(epoch) {
+			s, err = withdrawBalance(s, index, balance)
+			if err != nil {
+				return nil, errors.Wrap(err, "could not withdraw validator balance")
+			}
+			count++
+		} else if val.IsPartiallyWithdrawable(balance) {
+			s, err = withdrawBalance(s, index, balance-params.BeaconConfig().MaxEffectiveBalance)
```
Comment (saolyn) [inline beacon-chain/core/capella/transition.go:92]: It looks like you accidentally changed `params` to `balance-params` here

**Comment by potuz** [comment]
Closing since epoch transition will not change in the new design

**Pull request closed without merging**